commercelayer/validation.go

```diff
@@ -30,3 +30,27 @@ var inventoryModelStrategyValidation = func(i interface{}, path cty.Path) diag.D
 	return diag.Errorf("Invalid inventory model strategy provided: %s. Must be one of %s",
 		i.(string), strings.Join(getInventoryModelStrategies(), ", "))
 }
+
+func getPaymentSources() []string {
+	return []string{
+		"AdyenPayment",
+		"BraintreePayment",
+		"CheckoutComPayment",
+		"CreditCard",
+		"ExternalPayment",
+		"KlarnaPayment",
+		"PaypalPayment",
+		"StripePayment",
+		"WireTransfer",
+	}
+}
+
+var paymentSourceValidation = func(i interface{}, path cty.Path) diag.Diagnostics {
```

Comment:
Can you maybe add a test to check if we return an error if `Adyen` is provided? And one where we don't return an error (as the baseline). That way we also have the bug covered we ran into